core/web/evm_transfer_controller.go

```diff
@@ -107,3 +128,32 @@ func ValidateEthBalanceForTransfer(c *gin.Context, chain evm.Chain, fromAddr com
 
 	return nil
 }
+
+func FindTxAttempt(ctx context.Context, timeout time.Duration, etx txmgr.Tx, FindTxWithAttempts func(int64) (txmgr.Tx, error)) (attempt txmgr.TxAttempt, err error) {
+	recheckTime := time.Second
+	tick := time.After(0)
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	for {
+		select {
+		case <-ctx.Done():
+			return attempt, fmt.Errorf("%w - tx may still have been broadcast", ctx.Err())
+		case <-tick:
+			etx, err = FindTxWithAttempts(etx.ID)
+			if err != nil {
+				return attempt, fmt.Errorf("failed to find transaction: %w", err)
+			}
+		}
+
+		// exit if tx attempts are found
+		if len(etx.TxAttempts) > 0 {
+			break
+		}
+		tick = time.After(recheckTime)
+	}
```

Comment:
Could save some lines if this format is readable :shrug: 
```suggestion
	ctx, cancel := context.WithTimeout(ctx, timeout)
	defer cancel()
	// until attempts are found or errors encountered
	for tick := time.After(0); len(etx.TxAttempts) == 0; tick = time.After(recheckTime) {
		select {
		case <-ctx.Done():
			return attempt, fmt.Errorf("%w - tx may still have been broadcast", ctx.Err())
		case <-tick:
			etx, err = FindTxWithAttempts(etx.ID)
			if err != nil {
				return attempt, fmt.Errorf("failed to find transaction: %w", err)
			}
		}
	}
```

Reply:
it's definitely more condensed - but it feels a little harder to understand. you really have to think about the different parts of the `for` loop to understand what it's checking for compared to a very obvious exit check?